core/iwasm/common/gc/gc_common.c

```diff
@@ -79,16 +139,40 @@ wasm_func_type_get_result_type(WASMFuncType *const func_type, uint32 param_idx)
 uint32
 wasm_struct_type_get_field_count(WASMStructType *const struct_type)
 {
-    /* TODO */
-    return 0;
+    return struct_type->field_count;
 }
 
 wasm_ref_type_t
 wasm_struct_type_get_field_type(WASMStructType *const struct_type,
                                 uint32 field_idx, bool *p_is_mutable)
 {
     wasm_ref_type_t ref_type = { 0 };
-    /* TODO */
+    WASMStructFieldType field;
+    uint32 i;
+
+    bh_assert(field_idx < struct_type->field_count);
```

Comment:
Had better add `bh_assert(struct_type->type_flag == WASM_TYPE_STRUCT);` before this line